From dc98def875cd837ca92da852c24c5667e03d79b6 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Mon, 26 Aug 2013 16:10:05 -0700
Subject: [PATCH 253/382] MIPS: OCTEON: Fix section missmatch warnings in smp.c

Based on SDK octeon3_3.10.

The last commit caused some warnings due to poorly named variables.
Also add some 'static' and remove bogus 'inline'.

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/mips/cavium-octeon/smp.c |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/arch/mips/cavium-octeon/smp.c b/arch/mips/cavium-octeon/smp.c
index a0ce3cf..cfb1a25 100644
--- a/arch/mips/cavium-octeon/smp.c
+++ b/arch/mips/cavium-octeon/smp.c
@@ -478,7 +478,7 @@ late_initcall(register_cavium_notifier);
 
 #endif	/* CONFIG_HOTPLUG_CPU */
 
-struct plat_smp_ops octeon_smp_ops = {
+static struct plat_smp_ops octeon_smp_ops = {
 	.send_ipi_single	= octeon_send_ipi_single,
 	.send_ipi_mask		= octeon_send_ipi_mask,
 	.init_secondary		= octeon_init_secondary,
@@ -533,7 +533,7 @@ static void octeon_78xx_prepare_cpus(unsigned int max_cpus)
 	}
 }
 
-static inline void octeon_78xx_send_ipi_single(int cpu, unsigned int action)
+static void octeon_78xx_send_ipi_single(int cpu, unsigned int action)
 {
 	int i;
 
@@ -544,8 +544,8 @@ static inline void octeon_78xx_send_ipi_single(int cpu, unsigned int action)
 	}
 }
 
-static inline void octeon_78xx_send_ipi_mask(const struct cpumask *mask,
-					     unsigned int action)
+static void octeon_78xx_send_ipi_mask(const struct cpumask *mask,
+				      unsigned int action)
 {
 	unsigned int cpu;
 
@@ -553,7 +553,7 @@ static inline void octeon_78xx_send_ipi_mask(const struct cpumask *mask,
 		octeon_78xx_send_ipi_single(cpu, action);
 }
 
-struct plat_smp_ops octeon_smp_ops_78xx = {
+static struct plat_smp_ops octeon_78xx_smp_ops = {
 	.send_ipi_single	= octeon_78xx_send_ipi_single,
 	.send_ipi_mask		= octeon_78xx_send_ipi_mask,
 	.init_secondary		= octeon_init_secondary,
@@ -573,7 +573,7 @@ void __init octeon_setup_smp(void)
 	struct plat_smp_ops *ops;
 
 	if (OCTEON_IS_MODEL(OCTEON_CN78XX))
-		ops = &octeon_smp_ops_78xx;
+		ops = &octeon_78xx_smp_ops;
 	else
 		ops = &octeon_smp_ops;
 
-- 
1.7.0.4

